employee-rostering-backend/src/test/java/org/optaweb/employeerostering/spot/SpotServiceTest.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.spot;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.transaction.Transactional;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.optaweb.employeerostering.domain.skill.Skill;
+import org.optaweb.employeerostering.domain.spot.Spot;
+import org.optaweb.employeerostering.service.SkillService;
+import org.optaweb.employeerostering.service.SpotService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.web.util.NestedServletException;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+public class SpotServiceTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(SpotServiceTest.class);
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private SpotService spotService;
+
+    @Autowired
+    private SkillService skillService;
+
+    private Skill createSkill(Integer tenantId, String name) {
+        Skill skill = new Skill(tenantId, name);
+        Skill out = skillService.createSkill(tenantId, skill);
```

Comment:
Code style: redundant local variable. "Adds nothing to comprehensibility of the method", as IDEA reports :slightly_smiling_face:. You can return the value immediately.